src/test/bech32_tests.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017 Pieter Wuille
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "bech32.h"
+#include "test/test_bitcoin.h"
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(bech32_tests, BasicTestingSetup)
+
+bool CaseInsensitiveEqual(const std::string &s1, const std::string &s2)
+{
+    if (s1.size() != s2.size()) return false;
+    for (size_t i = 0; i < s1.size(); ++i) {
+        char c1 = s1[i];
+        if (c1 >= 'A' && c1 <= 'Z') c1 -= ('A' - 'a');
+        char c2 = s2[i];
+        if (c2 >= 'A' && c2 <= 'Z') c2 -= ('A' - 'a');
+        if (c1 != c2) return false;
+    }
+    return true;
+}
```

Comment:
Suggest using `std::tolower()` and standard algorithms to simplify the implementation

Reply:
Unfortunately, `tolower` is locale-dependent. I don't feel that the complexity of avoiding that is worth it for such a trivial algorithm.

Reply:
Locales scare me. As a reviewer, I would be far less confident that that code is correct due to it as opposed to an absolutely trivial algorithm that obvious does what it should.

Reply:
Agree. Seeing "locale" always means a non-trivial amount of googling is in store.

Reply:
Ok, makes sense. In that case I adding a utility function for 'english' [is|to][lower|upper], there's too much repetition of the same (simple) functionality.

Reply:
Yes, that really doesn't help. Please don't introduce dependence on locale. We prefer deterministic, well-defined, nation-independent string processing, even if that means writing some extra code at least we **know** what it's doing then. Also this is specific to bechs32 (part of the standard) so it's better to keep it self-contained.

Reply:
@laanwj Just to repeat an earlier comment, `<locale>` has overloads of these functions where you can specify the locale, so there are no hidden dependencies on the default system locale. I do agree it might require more googling, and is perhaps a bit less obvious.